### PR TITLE
Subdomain registration and certificate issuing

### PR DESCRIPTION
**Pull request by andrenatal**
@dhylands could you please review this?

### CODE REVIEW

**Comment by dhylands** [comment]
It looks like its failing in Travis (click on the red-X for details)

**Comment by andrenatal** [comment]
r=dhylands

File: app.js [hunk 1]
```diff
@@ -114,6 +139,12 @@ server.listen(port, function() {
   console.log('Listening on port', server.address().port);
 });
 
+// function to stop running server and start https
+TunnelService.switchToHttp = function(){
```
Comment (benfrancis) [inline app.js]: Nit: shouldn't this by switchToHttps?

File: config/default.js [hunk 2]
```diff
@@ -47,5 +48,11 @@ module.exports = {
     enabled: true,
     defaultUser: null,
     secret: 'top secret 51'      // DO NOT USE THIS IN PRODUCTION
+  },
+  ssltunnel: {
+    enabled: true,
+    registration_endpoint: 'mozilla-iot.org',
+    domain: 'box.mozilla-iot.org',
```
Comment (benfrancis) [inline config/default.js]: Why is this box.mozilla-iot.org ? It should be mozilla-iot.org. Was there a reason you weren't able to remove the box prefix?

File: views/tunnel_setup.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Mozilla IoT Gateway Subdomain Creation</title>
```
Comment (benfrancis) [inline views/tunnel_setup.html]: Nit: We changed all the other page titles to "Things Gateway" rather than "MozIoT Gateway" or "Mozilla IoT Gateway"

**Comment by benfrancis** [comment]
Just for reference, the UI mockup is here (there are several inconsistencies) https://docs.google.com/presentation/d/1rMeg6hep8JQflqwbiNIbhSoK9StCeVZ2npphmQHKda4/edit#slide=id.g221d7d7c9f_0_0

**Pull request merged**